components/supervisor/pkg/serverapi/publicapi.go

```diff
@@ -339,10 +339,11 @@ func (s *Service) publicAPIInstanceUpdate(ctx context.Context, errChan chan erro
 	for {
 		resp, err := resp.Recv()
 		if err != nil {
-			if err != io.EOF && status.Code(err) != codes.Unavailable {
+			code := status.Code(err)
+			if err != io.EOF && code != codes.Unavailable && code != codes.Canceled {
```

Comment:
@jeanp413 are you sure it works, I tried the same approach in https://github.com/gitpod-io/gitpod/pull/15795 

but it did not work, it seems that workspacekit wrap cancelled in unknown, i.e. it looks like cancelled in logs, but i practice it is status.Unknown.

I ended up doing `if err != io.EOF && ctx.Err() == nil && status.Code(err) != codes.Unavailable {` and it was enough

Reply:
Will check :+1: